provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go

```diff
@@ -0,0 +1,151 @@
+	resp.Schema = schema.Schema{
+		Description: "List of trusted ip addresses",
+		Attributes: map[string]schema.Attribute{
+			"page": schema.Int64Attribute{
+				Description: "Page number of the result set.",
+				Computed:    true,
+			},
+			"size": schema.Int64Attribute{
+				Description: "Number of items per page.",
+				Computed:    true,
+			},
+			"total": schema.Int64Attribute{
+				Description: "Total number of items in the result set.",
+				Computed:    true,
+			},
+			"items": schema.ListNestedAttribute{
+				Description: "Items of the list.",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description: "Ip address.",
+							Computed:    true,
+						},
+						"enabled": schema.BoolAttribute{
+							Description: "Indicates if the IP is enabled.",
+							Computed:    true,
+						},
+					},
+				},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func (s *TrustedIpsDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured:
+	if req.ProviderData == nil {
+		return
+	}
+
+	// Cast the provider data to the specific implementation:
+	connection := req.ProviderData.(*sdk.Connection)
+
+	// Get the collection of cloud providers:
+	s.collection = connection.ClustersMgmt().V1().TrustedIPAddresses()
+}
+
+func (s *TrustedIpsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	// Fetch the complete list of trusted IP addresses:
+	var listItems []*cmv1.TrustedIp
+	listSize := 10
+	listPage := 1
+	listRequest := s.collection.List().Size(listSize)
+
+	// Initialize variables to track total items and current page
+	var totalItems int
+	var currentPage int
+
+	for {
+		listResponse, err := listRequest.SendContext(ctx)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Can't list trusted IP addresses",
+				err.Error(),
+			)
+			return
+		}
+
+		// Update total items and current page values
+		totalItems = listResponse.Total()
+		currentPage = listResponse.Page()
+
+		if listItems == nil {
+			listItems = make([]*cmv1.TrustedIp, 0, totalItems)
+		}
+		listResponse.Items().Each(func(listItem *cmv1.TrustedIp) bool {
+			listItems = append(listItems, listItem)
+			return true
+		})
+		if listResponse.Size() < listSize {
+			break
+		}
+		listPage++
+		listRequest.Page(listPage)
+	}
+
+	// Populate the state:
+	state := &TrustedIpsState{
+		Page:  basetypes.NewInt64Value(int64(currentPage)),
+		Size:  basetypes.NewInt64Value(int64(listSize)),
+		Total: basetypes.NewInt64Value(int64(totalItems)),
+		Items: make([]*TrustedIpState, 0),
+	}
+
+	for _, listItem := range listItems {
+		trustedIpState := &TrustedIpState{
+			ID:      basetypes.NewStringValue(listItem.ID()),
+			Enabled: basetypes.NewBoolValue(listItem.Enabled()),
+		}
+
+		state.Items = append(state.Items, trustedIpState)
+	}
```

Comment:
This is a convert action between OCM object to tf object
Maybe it more appropriate to move this to its own function
Something like:
`trustedIpState := fromOCMListItems(listItems)`